api.md

```diff
@@ -0,0 +1,392 @@
+<div align="center">
+
+# Uploading files in `Phoenix` through an API
+
+Learn how to do image uploads from a `Phoenix` API!
+
+</div>
+
+> Before reading this guide,
+> please follow the instructions in [`README.md`](./README.md),
+> as this guide *continues* the progress made on it.
+
+
+# 1. Add `/api` scope and pipeline and setting up controller
+
+Let's create our API endpoint.
+Open `lib/router.ex`
+and uncomment the `pipeline :api` and `scope :api`.
+We are going to set our endpoint to create a given image
+that is sent via `multipart/form-data`.
+
+```elixir
+
+  pipeline :api do
+    plug :accepts, ["json"]
+  end
+
+  scope "/api", AppWeb do
+    pipe_through :api
+
+    resources "/images", ApiController, only: [:create]
+  end
+```
+
+We need to create our `ApiController`
+to serve these requests.
+Inside `lib/app_web/controllers`,
+create `api_controller.ex` 
+and paste the following code.
+
+```elixir
+defmodule AppWeb.ApiController do
+  use AppWeb, :controller
+  alias App.Todo
+
+  def create(conn, params) do
+    render(conn, :create)
+  end
+
+end
+```
+
+We're yet to serve anything, 
+we'll do this at a later stage.
+
+To render a `json` response,
+let's create a simple JSON template.
+In the same folder, create `api_json.ex`.
+
+```elixir
+defmodule AppWeb.ApiJSON do
+  def render("create.json", _assigns) do
+    %{url: "Some URL"}
+  end
+
+  def render(template, _assigns) do
+    %{errors: %{detail: Phoenix.Controller.status_message_from_template(template)}}
+  end
+end
+```
+
+Now, depending on the status of the response,
+we will render a sample response
+`%{url: "Some URL"}`.
+Don't worry, 
+we'll change this with the public URL
+after implementing the feature 
+that uploads the image file to `S3`.
+
+
+# 2. Uploading to `S3` bucket
+
+In order to upload the file to an `S3` bucket,
+we are going to make use of the 
+[`ex_aws`](https://github.com/ex-aws/ex_aws) package.
+Let's install it by adding the following lines 
+to the `deps` section in `mix.exs`.
+
+```elixir
+      {:ex_aws, "~> 2.0"},
+      {:ex_aws_s3, "~> 2.0"},
+      {:sweet_xml, "~> 0.7"}
+```
+
+Run `mix deps.get` to download the dependencies.
+
+Next, we need to add configuration
+of this newly added dependencies
+in `config/config.ex`.
+Open it and add these lines.
+
+```elixir
+config :ex_aws,
+  access_key_id: System.fetch_env!("AWS_ACCESS_KEY_ID"),
+  secret_access_key: System.fetch_env!("AWS_SECRET_ACCESS_KEY"),
+  region: "eu-west-3"
```

Comment:
`region` shouldn't be hard-coded. 
but in _our_ case it can be, for now ...

Reply:
Yeah, I'm sure I think it makes sense to have this over as an env variable. I'll change this 👍